### PR TITLE
Feature/rspec api documentation integration

### PR DESCRIPTION
**Pull request by JuannFerrari**
#### Board:
* [ISSUE](https://github.com/rootstrap/rails_api_base/issues/266)
---
#### Description:

This PR adds `rspec_api_documentation` gem and all API endpoint acceptance specs to help us to auto-document it. IMO having this gem and the specs as an example will help us to encourage acceptance specs in the projects that use this base as a template, giving good documentation about their APIs.

---
#### Notes:
* `aws-sdk-s3', '~> 1.75` changed in gemfile to make `bundle install` works
* `RspecApiDocumentation` initializer has monkey patching to prevent an issue with Rails 6. (this has been already fixed in https://github.com/zipmark/rspec_api_documentation/issues/456 and I'm waiting for a new release of the gem)
* `apiary.apib` is now generated by `./bin/docs` task

### CODE REVIEW

File: config/initializers/rspec_api_documentation.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+module RspecApiDocumentation
```
Comment (juanmanuelramallo) [inline config/initializers/rspec_api_documentation.rb:1]: Shouldn't this be on the same unless block below?

**Review by juanmanuelramallo — COMMENTED** [review]
Loved it! 

What do you think of adding something like this https://github.com/juanmanuelramallo/target-mvd/blob/master/spec/support/devise_token_auth_overrides.rb to make the API Blueprint document always use the same access token and client id, just to avoid unnecessary changes in the document (?)

File: config/initializers/rspec_api_documentation.rb (hunk 1 above)
Comment (JuannFerrari) [reply config/initializers/rspec_api_documentation.rb:1]: Sure! and it's a nice idea to override that tokens indeed, I will add it.

File: README.md [hunk 2]
```diff
@@ -103,6 +105,15 @@ This template comes with:
 
 http://docs.railsapibase.apiary.io
 
+With [Rspec API Doc Generator](https://github.com/zipmark/rspec_api_documentation) you can generate the docs after writing the acceptance specs.
+
+Just run:
+
+` sh ./bin/docs `
```
Comment (juanmanuelramallo) [inline README.md:112]: Do we need the `sh` at the beginning? 

```suggestion
`./bin/docs `
```

File: README.md [hunk 3]
```diff
@@ -103,6 +105,15 @@ This template comes with:
...
+
+And an `apiary.apib` file will be generated.
```
Comment (juanmanuelramallo) [inline README.md:114]:
```suggestion
An `apiary.apib` file will be generated at the root directory of the project.
```

File: config/initializers/rspec_api_documentation.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
+unless Rails.env.production?
+
```
Comment (juanmanuelramallo) [inline config/initializers/rspec_api_documentation.rb:2]: nitpick: unneeded empty newline (?)

**Review by juanmanuelramallo — APPROVED** [review]
simply B E A U T I F U L

File: README.md (hunk 2 above)
Comment (JuannFerrari) [reply README.md:112]: At least for zsh shell, yes 🤔

**Review by brunvez — APPROVED** [review]

**Pull request merged**